---
title: "GC Analysis of Chromosome 1"
author: "Alisha Kalangara"
output:
  pdf_document: default
header-includes:
- \usepackage{booktabs}
- \usepackage{float}
---

```{r setup, include=FALSE, cache=TRUE}
knitr::opts_chunk$set(echo = TRUE, fig.pos='H', fig.show = 'hold')

#download data
url = "http://www.haowulab.org/teaching/bioc/hg19_CGI.txt"
download.file(url,dest=basename(url), quiet = FALSE, mode = "w", cacheOK = TRUE)

#install bioconductor
if (!requireNamespace("BiocManager")) 
     install.packages("BiocManager") 

#install bioconductor packages
BiocManager::install()
BiocManager::install(c("GenomicRanges", "GenomicFeatures", "BSgenome", "BSgenome.Hsapiens.UCSC.hg19", "Biostrings"))

#install R packages
installed_pkgs <- row.names(installed.packages())
pkgs <- c("tinytex","kableExtra", "tidyverse","devtools")
for(p in pkgs){
	if(!(p %in% installed_pkgs)){
		install.packages(p, repos = "http://cran.us.r-project.org")
	}
}

#load packages
library(tinytex)
library(Biostrings)
library(BSgenome)
library(BSgenome.Hsapiens.UCSC.hg19)
library(GenomicFeatures)
library(kableExtra)
library(tidyverse)
library(devtools)
```
# Background Information  
It is known that in many mammalian genomes, C and G base pairs do not appear as frequently as A and T base pairs. Additionally, C and G base pairs also tend to not be found next to each other. DNA methylation has been posited as a possible reason for this occurrence as when Cs and Gs occur together, C are methylated and resemble Ts in their chemical structure. Consequently, as DNA replicates, a mutation may occur such that CGs are turned into TGs. Wherever CGs have remained in genomes are denoted as CpG islands as these locations indicate important regions of the genome as these areas have been conserved over time.  
  
# Methods
To investigate this idea, we focused on the 1st chromosome of the human genome. Initially the GC content of the 1st chromosome was determined to see whether it was low, as other mammalian genomes suggest.

To assess whether DNA methylation could be responsible for a low GC content in mammals, we compare the observed GC dinucleotide occurence to the number of times they are expected to be together. This observed to expected ratio should be around 1 if Cs and Gs occur independently of one another. Additionally, if CGs are being replaced by TGs, TG should be be have a high observed to expected ratio.

To determine the expected values, we compute the frequency of each of the base pairs in the chromosome. The expected value of a dinucleotide is the product of the individual frequencies times the total number of basepairs.

To further take a look at the CG content and presence of CpG islands, we also graphed the frequency CG% and CG OER in each of 1000 bp long intervals of chr 1.  

# Results & Conclusions
```{r, echo=FALSE, cache=TRUE}
## get sequence for chromosome 1
Seq=Hsapiens[["chr1"]]

## determine get GC content
bases=alphabetFrequency(Seq,baseOnly=TRUE)
ntotBases=sum(bases[1:4])
baseFreq=(bases[1:4]/ntotBases)
GCcontent=(baseFreq["C"]+baseFreq["G"])*100

## look at CG dinucleotide content
find_cg=matchPattern("CG", Seq)
num_cg=length(find_cg)
oer_cg= num_cg/(baseFreq["C"]*baseFreq["G"]*ntotBases)

## loot at TG dinucleotide content
find_tg=matchPattern("TG", Seq)
num_tg=length(find_tg)
oer_tg=num_tg/(baseFreq["T"]*baseFreq["G"]*ntotBases)

## GC content in 1000bp windows
ss=seq(1, length(Seq), by=1000)
ss=ss[-length(ss)] ## remove the last one
Seq.set=DNAStringSet(Seq, start=ss, end=ss+999)
ff=alphabetFrequency(Seq.set, baseOnly=TRUE)
pCG=((ff[,"C"]+ff[,"G"])/rowSums(ff))*100

## CG occurance in 1000bp windows
Seq.set=DNAStringSet(Seq, start=ss, end=ss+999)
nCG=vcountPattern("CG", Seq.set)
oer=nCG*1000/(ff[,"C"]*ff[,"G"])
```
For chromosome 1 of the human genome, the GC content was found to be `r round(GCcontent,digits=0)`%. This is lower than 50% and shows that at least for chromosome 1, humans do have a low GC content.

In order to determine the expected values we compute the frequency of each of the base pairs in the chromosome shown in Table 1. We see that the frequencies are not distributed evenly and that As and Ts are more frequent.

In order to determine the observed to expected ratio, we first compute each part separately and then divide them. The expected value of a dinucleotide is the product of the individual frequencies times the total number of basepairs.

In order to determine the observed values, we simply counted the
the number of times Cs and Gs appear in the chromosome together and Ts and Gs appear in the chromosome together.

Table 2 shows the Observed to Expected Ratios for CGs compared to TGs. We see that the OER for CG is less than 1 indicating that Cs and Gs are not independent of one another. The same can be said of Ts and Gs as the OER is also not 1. Comparing the OERs for CGs and TGs, CGs appear less than expected (OER is less than 1), while TGs appear more than expected (OER greater than 1).  
```{r, echo=FALSE, results='asis'}
T1 <- matrix(round(baseFreq,digits=2), ncol=4, byrow=FALSE)
T1 <- as.data.frame(T1)

T2 <-matrix(round(c(oer_cg,oer_tg),digits=2), ncol=1, byrow=FALSE)
rownames(T2) <- c("CG", "TG")
T2 <- as.data.frame(T2)

TT1 <- kable(T1, format = "latex", col.names = c("A", "C", "G", "T"), booktabs = TRUE)%>%
  kable_styling(latex_options = "HOLD_position")

TT2 <- kable(T2, format = "latex", col.names = c("OER"),booktabs = TRUE)%>%
  kable_styling(latex_options = "HOLD_position")


cat(c("\\begin{table}[H]
    \\begin{minipage}{.5\\linewidth}
      \\caption{Nucleotide Frequencies in Chr 1}
      \\centering",
        TT1,
    "\\end{minipage}%
    \\hspace{-1.25cm}
    \\begin{minipage}{.7\\linewidth}
      \\centering
        \\caption{Observed to Expected Ratios for CG vs TG}",
        TT2,
    "\\end{minipage} 
\\end{table}"
)) 
```

Lastly, to view the GC content and CpG island presence across the chromosome, we evaluated the GC content in each 1000 bp intervals. The CpG presence was denoted by a high observed to expected ratio which was also calculated in each of the 1000bp intervals.  

```{r, echo=FALSE, fig.cap= "Frequencies of GC content for 1000bp intervals of Chr 1",}
hist(pCG[pCG>0],100, xlab="Percent CG", main="")
```
  
```{r, echo=FALSE, fig.cap= "Frequencies of GC OER for 1000bp intervals of Chr 1"}
hist(oer,100, xlab = "GC OER", main="")
```
From Figure 1. we see that the GC content is fairly low for most intervals but is high in a few intervals. From Figure 2. we see that the presence of CpG islands is lower than expected for most intervals but the tail on the histogram suggests the presence of a several intervals where CG content was high.  

# Trouble Shooting Info
```{r,echo=TRUE}
getwd()
devtools::session_info()
```

